### ISSUE TITLE
Memory settings

### ISSUE THREAD

**Issue by ellaerror** [TobiasM95/WoW-Talent-Tree-Manager#20]
Hello, this is a great tool. I had some suggestions (mostly regarding resource utilization) I hope are helpful. 
I'm using the release build v1.3.8.

- It would be cool if you could change your max memory or max solutions to allow for more options, I have 32GB of ram on my computer and I only saw 35% overall ram utilization when processing one tree. However, it still canceled the solver because there were too many options. Settings could be a dialog in the program or parsed from a text file. 
- As a solution for users with more limited ram, you could potentially write 'checkpoints' to a disk file. kind of like a swap file. It would obviously be much slower, but some people might not mind waiting a very long time. 
- Once the class trees are more nailed down, you could create a "solutions" file or something that comes pre-downloaded so you don't have to generate all the options.
- I noticed you can exclude nodes from the solver (via filtering), but only after generating all the solutions. Allowing people to exclude them beforehand could reduce the number of solutions by a lot.
- It would be nice to have the loadout solver and the loadout editor side by side so you don't have to keep swapping tabs when checking out the different builds. 

I would recompile with some different limits or something but I'm clueless when it comes to visual studio and windows development. If you need any lightweight libraries in C though I might be able to help contribute, although I haven't looked at the source code.

I downloaded this to see how many ways you could use 8 points in the havoc class tree and I was not disappointed. (btw, the answer is 375).

**Comment by TobiasM95** [comment]
I decided on a dynamic safety guard size. 4 GiB of RAM are reserved and will not be used by TTM, rest is fair game. Should lead to maximum RAM usage for a 32 GB system of about 28 GB. Will be included in the next update.
